tests/FoliageSpec.hs

```diff
@@ -0,0 +1,12 @@
+module FoliageSpec where
+
```

Comment:
I feel dumb here: I struggle a bit to import `Foliage` so I can write tests :/

Reply:
The problem is that everything was previously defined in an executable. You want to put it in a library so you can import it.